packages/cli/src/metadataGeneration/typeResolver.ts

```diff
@@ -303,6 +303,87 @@ export class TypeResolver {
       return new TypeResolver(this.typeNode.type, this.current, this.typeNode, this.context, this.referencer).resolve();
     }
 
+    if (ts.isTypeQueryNode(this.typeNode)) {
+      const isIntrinsicType = (type: ts.Type): boolean => {
+        const flags = ts.TypeFlags;
+        return (
+          this.hasFlag(type, flags.Number) ||
+          this.hasFlag(type, flags.String) ||
+          this.hasFlag(type, flags.Boolean) ||
+          this.hasFlag(type, flags.BigInt) ||
+          this.hasFlag(type, flags.ESSymbol) ||
+          this.hasFlag(type, flags.Undefined) ||
+          this.hasFlag(type, flags.Null)
+        );
+      };
+      const symbol = this.current.typeChecker.getSymbolAtLocation(this.typeNode.exprName);
+      if (symbol) {
+        // Access the first declaration of the symbol
+        const declaration = symbol.declarations?.[0];
+        throwUnless(declaration, new GenerateMetadataError(`Could not find declaration for symbol: ${symbol.name}`, this.typeNode));
+
+        if (ts.isVariableDeclaration(declaration)) {
+          const initializer = declaration.initializer;
+          const declarationType = this.current.typeChecker.getTypeAtLocation(declaration);
+          if (isIntrinsicType(declarationType)) {
+            return { dataType: this.current.typeChecker.typeToString(declarationType) as Tsoa.PrimitiveType['dataType'] };
+          }
+          if (initializer && (ts.isStringLiteral(initializer) || ts.isNumericLiteral(initializer) || ts.isBigIntLiteral(initializer))) {
+            return { dataType: 'enum', enums: [initializer.text] };
+          } else if (initializer && ts.isObjectLiteralExpression(initializer)) {
+            const getOneOrigDeclaration = (prop: ts.Symbol): ts.Declaration | undefined => {
+              if (prop.declarations) {
+                return prop.declarations[0];
+              }
+              const syntheticOrigin: ts.Symbol = (prop as any).links?.syntheticOrigin;
+              if (syntheticOrigin && syntheticOrigin.name === prop.name) {
+                return syntheticOrigin.declarations?.[0];
+              }
+              return undefined;
+            };
+
+            const isIgnored = (prop: ts.Symbol) => {
+              const declaration = getOneOrigDeclaration(prop);
+              return declaration !== undefined && getJSDocTagNames(declaration).some(tag => tag === 'ignore') && !ts.isPropertyAssignment(declaration);
+            };
+
+            const typeProperties: ts.Symbol[] = this.current.typeChecker.getPropertiesOfType(this.current.typeChecker.getTypeAtLocation(initializer));
+            const properties: Tsoa.Property[] = typeProperties
```

Comment:
A lot of the property management should already exist in checker based resolutions.

Ideally we can merge them together.

Reply:
Can you elaborate on what you mean? Do you mean putting this code in a location that can be shared across other places where we do property management in this file?

Reply:
This is taken from lines 194ff, right?

Reply:
Yes, parts of it are taken from another section. Do you want me to extract that to a shared method is my question

Reply:
I haven't had time to spend on it lately. Happy for anyone else to contribute to clean up on this pr to get it across the line. It's functionally complete, but needs a little organizing